src/services/demoRequest.service.ts

```diff
@@ -0,0 +1,37 @@
+// Dao
+import DemoRequestDAO from '@/dao/demoRequest.dao';
+import SessionDAO from '@/dao/session.dao';
+// Helpers
+import EmailHelper from '@/helpers/email.helper';
+// Formatters
+import { DemoRequestFormatter } from '@/formatters/demoRequest.formatter';
+// Typings
+import { BookDemoRequestData } from '@/typings/demoRequest';
+
+class DemoRequestService {
+  // Dao
+  private demoRequestDAO = new DemoRequestDAO();
+  private sessionDAO = new SessionDAO();
+  // Helpers
+  private emailHelper = new EmailHelper();
+  // Formatters
+  private demoRequestFormatter = new DemoRequestFormatter();
+  /**
+   * Validate demo request and create new demo request entry in DB.
+   * @param {BookDemoRequestData} bookDemoRequestData
+   */
+  public bookDemoRequest = async (bookDemoRequestData: BookDemoRequestData) => {
+    // Format user data.
+    const formattedBookDemoRequestData = this.demoRequestFormatter.formatNewDemoRequestData(bookDemoRequestData);
+
+    // Create user entry in DB.
+    await this.demoRequestDAO.createDemoRequest(formattedBookDemoRequestData);
+
+    // Sending demo booking confirmation mail to visitor.
+    await this.emailHelper.sendBookDemoRequestEmail(bookDemoRequestData.email, bookDemoRequestData.prefferedDate);
+
+    return { message: 'Free demo booking confirmation sent to your email' };
```

Comment:
just send success or failure.
ui will handle it using withErrorHandler title: "Free Demo Booked Successfully".
subtitle/subheading "You will receive confirmation email shortly"

Reply:
beacuse FE will not store response of post request.

Reply:
Done